adsingestp/parsers/copernicus.py

```diff
@@ -0,0 +1,209 @@
+                self.base_metadata["publication"] = journal_metadata.find(
+                    "journal_title"
+                ).get_text()
+
+            if journal_metadata.find("volume_number"):
+                self.base_metadata["volume"] = journal_metadata.find("volume_number").get_text()
+
+    def _parse_pagination(self):
+        if self.input_metadata.find("start_page"):
+            self.base_metadata["page_first"] = self.input_metadata.find("start_page").get_text()
+
+        if self.input_metadata.find("end_page"):
+            self.base_metadata["page_last"] = self.input_metadata.find("end_page").get_text()
+
+        if self.record_meta.find("article_number"):
+            self.base_metadata["electronic_id"] = self.input_metadata.find(
+                "article_number"
+            ).get_text()
+
+    def _parse_ids(self):
+        self.base_metadata["ids"] = {}
+
+        issns = []
+        if self.input_metadata.find("issn"):
+            issns.append(("print", self.input_metadata.find("issn").get_text()))
+
+        if self.input_metadata.find("eissn"):
+            issns.append(("electronic", self.input_metadata.find("issn").get_text()))
+
+        self.base_metadata["issn"] = issns
+
+        if self.input_metadata.find("doi"):
+            self.base_metadata["ids"]["doi"] = self.input_metadata.find("doi").get_text()
+
+    def _parse_title(self):
+        title_array = self.input_metadata.find_all("article_title")
+        if title_array:
+            title_array_text = [i.get_text() for i in title_array]
+            # use BS to remove html markup inside title field
+            if len(title_array) == 1:
+                title_temp = BeautifulSoup(title_array_text[0], "html.parser")
+                title = title_temp.get_text().title()
+
+                self.base_metadata["title"] = title
+            else:
+                title_temp = BeautifulSoup(": ".join(title_array_text), "html.parser")
+                title = title_temp.get_text().title()
+
+                self.base_metadata["title"] = title
+
+        else:
+            raise MissingTitleException("No title found")
+
+    def _parse_author(self):
+        author_list = []
+        name_parser = utils.AuthorNames()
+
+        affil_map = {}
+
+        # Create a dictionary to map affiliation names to affiliation numbers
+        if self.input_metadata.find_all("affiliations"):
+            affil_list = self.input_metadata.find_all("affiliations")[0].find_all("affiliation")
+            for aff in affil_list:
+                affil_map[aff.get("numeration", "")] = self._clean_output(aff.get_text())
+
+        author_array = self.input_metadata.find_all("author")
+        for a in author_array:
+            author_temp = {}
+            name = a.find("name").get_text()
+            parsed_name = name_parser.parse(
+                name, collaborations_params=self.author_collaborations_params
+            )
+            author_temp = parsed_name[0]
```

Comment:
Is there ever more than one element in the `parsed_name` array? It might be safer to do a `for` loop over `parsed_name` and append all of those to `author_list`

Reply:
I have not come across more than one element in `parsed_name` in the current setup where it is inside a for loop cycling through each author individually. The reason for the first element being accessed is that name_parser returns a list of dictionaries of names, and I need to access the dictionary of the single name in the list. e.g.:
```
parsed_name
[{'given': 'Marie', 'middle': '', 'surname': 'Dumont', 'suffix': '', 'prefix': '', 'nameraw': 'Marie Dumont'}] 
```